packages/next/src/shared/lib/router/router.ts

```diff
@@ -569,10 +569,13 @@ function fetchNextData({
         })
       })
       .then((data) => {
+        const xMiddlewareCacheValues = data.response.headers.get('x-middleware-cache')
+        const isExistxMiddlewareCacheValue = xMiddlewareCacheValues.split(',')
+          .some(value => ['private', 'no-cache', 'no-store', 'max-age=0', 'must-revalidate'].includes(value))
```

Comment:
Thank you for the review! 
The `includes` function cannot be used immediately with the suggested `middlewareCacheHeader`.

> The includes() method of Array instances determines whether an array includes a certain value among its entries, returning true or false as appropriate.(MDN -[ Array.prototype.includes()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes))

So I think it would be right to use the `split` function to create an array of strings and use the `includes` function!

I added `trim` logic because I didn't process any additional spaces!